From 005603010bf67b37c5757a1e8f61165dc7684152 Mon Sep 17 00:00:00 2001
From: Elad Yifee <eladwf@gmail.com>
Date: Mon, 29 Apr 2024 16:41:11 +0300
Subject: [PATCH] net: ethernet: mtk_eth_soc: Fix DMA for devices with more
 than 4GB of dram
Add GFP_DMA32 flag to pages allocations in order to support devices with more
than 4GB of dram

Signed-off-by: Elad Yifee <eladwf@gmail.com>
---
 drivers/net/ethernet/mediatek/mtk_eth_soc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mediatek/mtk_eth_soc.c b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
index 0000000000..94c629a554
--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
@@ -1890,7 +1890,7 @@ static void *mtk_page_pool_get_buff(struct page_pool *pp, gfp_t gfp_mask)
 {
 	struct page *page;
 
-	page = page_pool_alloc_pages(pp, gfp_mask | __GFP_NOWARN);
+	page = page_pool_alloc_pages(pp, gfp_mask | __GFP_NOWARN | GFP_DMA32);
 	if (!page)
 		return NULL;
